### PR TITLE
Move Domain Connect Sync Flow from React PropTypes to prop-types

### PR DESCRIPTION
**Pull request by delputnam**
This PR is just migrating the PropTypes from the React PropTypes to PropTypes from prop-types.

 D5862-code (which is now merged), includes instructions on how to test the Domain Connect Sync Flow.

### CODE REVIEW

File: client/my-sites/domains/domain-management/domain-connect/domain-connect-authorize.jsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /**
  * External dependencies
- */
-import React, { Component, PropTypes } from 'react';
+*/
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/domain-connect/domain-connect-authorize.jsx]: OMG WHY DID YOU REMOVE THIS SPACE?!?!?!?!111 What does it all _mean_? 
....
:trollface: 
Seriously, though, probably best to keep it consistently formatted across files.

Comment (delputnam) [reply client/my-sites/domains/domain-management/domain-connect/domain-connect-authorize.jsx]: Huh. I definitely didn't mean to do that.  Will fix. Thanks!
